src/pipecat/services/aws/llm.py

```diff
@@ -790,6 +791,78 @@ def can_generate_metrics(self) -> bool:
         """
         return True
 
+    async def generate_summary(
+        self, summary_prompt: str, context: LLMContext | OpenAILLMContext
+    ) -> Optional[str]:
+        """Generate a conversation summary from the given LLM context.
+
+        Args:
+            summary_prompt: The prompt to use to guide generating the summary.
+            context: The LLM context containing conversation history.
+
+        Returns:
+            The generated summary, or None if generation failed.
+        """
+        try:
+            if isinstance(context, LLMContext):
+                # Not sure if it's strictly necessary to adapt messages here
+                # since they'll just be a string in the prompt, but erring on
+                # the side of putting them in the format the LLM would expect
+                # if consuming them directly (i.e. assuming greater LLM
+                # familiarity with its own format).
+                # adapter = self.get_llm_adapter()
+                # params: AWSBedrockLLMInvocationParams = adapter.get_llm_invocation_params(context)
+                # messages = params["messages"]
+                raise NotImplementedError(
+                    "Universal LLMContext is not yet supported for AWS Bedrock."
+                )
+            else:
+                messages = context.messages
+
+            # Determine if we're using Claude or Nova based on model ID
+            model_id = self.model_name
+
+            # Prepare request parameters
+            request_params = {
+                "modelId": model_id,
+                "messages": [
+                    {
+                        "role": "user",
+                        "content": [{"text": f"Conversation history: {messages}"}],
+                    },
+                ],
+                "inferenceConfig": {
+                    "maxTokens": 8192,
+                    "temperature": 0.7,
+                    "topP": 0.9,
+                },
+            }
+
+            request_params["system"] = [{"text": summary_prompt}]
+
+            # Call Bedrock without streaming
+            response = self._client.converse(**request_params)
```

Comment:
Ah, glad I'm doing this refactor! Discovered that Pipecat Flows is currently broken, since it reaches into `AWSBedrockLLMService` to grab this `_client` property, which actually went away with this change: https://github.com/pipecat-ai/pipecat/pull/2148